lib/associations/belongs-to.js

```diff
@@ -54,7 +54,7 @@ var BelongsTo = function(source, target, options) {
     !this.target.options.underscored
   );
 
-  this.targetIdentifier = this.target.primaryKeyAttribute;
+  this.targetIdentifier = this.options.targetKey || this.target.primaryKeyAttribute;
```

Comment:
Ah, I see what you mean. And then use that attribute in the `query-generator` code.  Sure, makes sense.  In the spirit of DRY, I'd rather not make two attributes that represent the same thing, but I'll try using `targetIdentifier` in the `query-generator` code. That should work (and be better).